src/spec-node/singleContainer.ts

```diff
@@ -160,14 +161,24 @@ async function buildAndExtendImage(buildParams: DockerResolverParameters, config
 
 	const args: string[] = [];
 	if (buildParams.buildKitVersion) {
-		args.push('buildx', 'build',
-			'--load', // (short for --output=docker, i.e. load into normal 'docker images' collection)
-			'--build-arg', 'BUILDKIT_INLINE_CACHE=1', // ensure cache manifest is included in the image
-		);
+		if (buildParams.buildxPlatform && buildParams.buildxPush && argImageName) {
+			args.push('buildx', 'build');
+			args.push('--platform', buildParams.buildxPlatform);
+			args.push('--push');
+			args.push('-f', finalDockerfilePath, '-t', argImageName);
+		} else {
+			args.push('buildx', 'build',
+				'--load', // (short for --output=docker, i.e. load into normal 'docker images' collection)
+				'--build-arg', 'BUILDKIT_INLINE_CACHE=1', // ensure cache manifest is included in the image
+			);
+		}
 	} else {
 		args.push('build');
 	}
-	args.push('-f', finalDockerfilePath, '-t', baseImageName);
+	if (!argImageName) {
```

Comment:
If this method just took a single image name (i.e. the `buildNamedImageAndExtend` used the image name it is passed and falls back to the previous `'image' in config ? config.image : getFolderImageName(params.common)` value if not set) then this if block wouldn't be needed. I think it would also allow the added code above to be more similar between the single and multi-platform options?

Reply:
@stuartleeks , thanks for the suggestion. Let me know if this is what you had in mind. PR updated.